### DISCUSSION TITLE
Mike Urban Plus - user written controls -SQLite

### DISCUSSION THREAD

**Discussion by lizaclark** [DHI/Mike1D-SDK-examples#17] [Q&A] [answered]
Thanks for providing this set of Mike1D SDK examples. We're transitioning to Mike Urban Plus and looking to transition our user-written controls to the new framework. I have noticed that the examples in this repo use the *.m1dx file as a basis for updating the model simulations. In my experience so far, Mike Urban Plus does not write an *.m1dx file but presumably works from the SQLite database directly. Do you know of any examples for interacting directly with Mike1D in the Mike Urban Plus framework? Is anyone working on this? Or does the DHI.M1DSimulator.M1DSimulatorRtc() function work with a SQLite file as well?

**Comment by JesperGr** [comment, marked as answer]
Hi Elizabeth

Both scripts and plugins are applicable for all sources of setup data, as long as MIKE 1D can consume the setup data. So, it will work for MIKE URBAN Classic (.mdb/.m1dx), MIKE URBAN+ (.mupp), MIKE HYDRO (.mhydro), MIKE 11 (.sim11+more), and also the "exported" MIKE 1D (.m1dx) data. Hence, any of the scripts in the [scripts](https://github.com/DHI/Mike1D-SDK-examples/tree/master/DHI.Mike1D.Examples/Scripts) folder can be used for any type of setup data.

Example, for MIKE URBAN+, if you have a MyModel.sqlite and a MyModel.mupp file, then you can make a MyModel.cs (script) or a MyModel.m1da (plugin), which will then be executed for all simulations (RR or HD or combined), regardless of whether the simulation run is initiated from the MIKE URBAN+ interface or from a command line. 

If the script should only be applied on some of the simulations in the database, it is possible to run MIKE 1D on the database directly, specifying which simulation id to run and which script to use, something like:

`"c:\Program Files (x86)\DHI\MIKE URBAN\2020\bin\x64\DHI.Mike1D.Application.exe" MyModel.mupp -simulationid=XYZ -script=scriptFile.cs`

Alternative, any type of data can be exported to MIKE 1D .m1dx. MIKE URBAN+ has a menu-item for doing this. It is also possible through command line:

`"c:\Program Files (x86)\DHI\MIKE URBAN\2020\bin\x64\DHI.Mike1D.Application.exe" MyModel.mupp -simulationid=XYZ -m1dx=MySimulation.m1dx`

which will run the export step and save the simulation to a MySimulation.m1dx. To see which options are available, try:

`"c:\Program Files (x86)\DHI\MIKE URBAN\2020\bin\x64\DHI.Mike1D.Application.exe" -help`

You may want to use a '-gui -close' or '-console' options...

Be aware - this comment is subject to errors: Check paths and exact spelling of arguments - I am writing this out of memory :-)
